pkg/phlaredb/symdb/symdb_test.go

```diff
@@ -227,3 +227,23 @@ func TestWritePartition(t *testing.T) {
 `
 	require.Equal(t, expected, resolved.String())
 }
+
+func BenchmarkPartitionWriter_WriteProfileSymbols(b *testing.B) {
+	b.ReportAllocs()
+
+	for i := 0; i < b.N; i++ {
+		// If opening the profile or creating the config, db, or
+		// partition writer outside the loop it always results in a
+		// panic.
+		b.StopTimer()
+		p, err := pprof.OpenFile("testdata/profile.pb.gz")
+		require.NoError(b, err)
+		p.Normalize()
+		cfg := DefaultConfig().WithDirectory(b.TempDir())
+		db := NewSymDB(cfg)
+		pw := db.PartitionWriter(0)
+		b.StartTimer()
+
+		pw.WriteProfileSymbols(p.Profile)
```

Comment:
During ingest we are actively mutate the profile, that's why you can't use the profile a 
second time. You can use the protobuf tooling to clone the parsed profile.

I also changed it to use a new partition for every iteration. Otherwise the stacktraces are completely existing and you will no longer ingest with the missing elements.

Ideally you need better input data to make this behave more like with real profiles.


```suggestion

	p, err := pprof.OpenFile("testdata/profile.pb.gz")
	require.NoError(b, err)
	p.Normalize()
	cfg := DefaultConfig().WithDirectory(b.TempDir())

	db := NewSymDB(cfg)

	for i := 0; i < b.N; i++ {
		b.StopTimer()
		newP := p.CloneVT()
		pw := db.PartitionWriter(uint64(i))
		b.StartTimer()

		pw.WriteProfileSymbols(newP)
```